packages/test-utils/src/utils.ts

```diff
@@ -47,6 +47,52 @@ export const waitForEvent = (emitter: EventEmitter, event: Event, timeout = 5000
     })
 }
 
+/**
+ * Run functions and wait for events to be emitted within timeout. Returns a promise created with Promise.all() 
+ * and waitForEvent() calls. Calls the functions after creating the promise.
+ *
+ * @param operations function(s) to call
+ * @param waitedEvents event(s) to wait for
+ * @param timeout amount of time in milliseconds to wait for
+ * @returns {Promise<unknown[]>} resolves with event arguments if event occurred
+ * within timeout. Otherwise rejected.
+ */
+export const runAndWaitForEvents = (
+    operations: () => void | (() => void)[], 
+    waitedEvents: [emitter: EventEmitter, event: Event] | Array<[emitter: EventEmitter, event: Event]>, 
+    timeout = 5000
+): Promise<unknown[]> => {
+    let ops: (() => void)[]
+    if (Array.isArray(operations)) {
+        ops = operations
+    } 
+    else {
+        ops = [operations]
+    }
+
+    let evs: Array<[emitter: EventEmitter, event: Event]>
+    if (Array.isArray(waitedEvents) && Array.isArray(waitedEvents[0]) ) {
+        evs = waitedEvents as Array<[emitter: EventEmitter, event: Event]>
+    } 
+    else {
+        evs = [waitedEvents as [emitter: EventEmitter, event: Event]]
+    }
+    
+    const promise: Promise<unknown[]> = new Promise( (resolve, reject) => { 
+        Promise.all(evs.map( (e) => { waitForEvent(e[0], e[1], timeout) }))
```

Comment:
Ah there's a critical bug here. The promise returned by `waitForEvent` isn't being returned due to the `{` in `(e) => { waitForEvent(…) }`, so this is essentially just creating an array of undefined and a bunch of unchained async. The `Promise.all` won't actually wait for anything here.

Can fix it like so, also explicitly named the arguments:

```suggestion
`const promise = Promise.all(evs.map(([emitter, event]) => waitForEvent(emitter, event, timeout)))`
```